UnityProject/Assets/Scripts/Items/Wieldable.cs

```diff
@@ -0,0 +1,180 @@
+using Items;
+using Messages.Server;
+using Mirror;
+using UnityEngine;
+
+namespace Items
+{
+	public class Wieldable : NetworkBehaviour, IServerInventoryMove, ICheckedInteractable<HandActivate>
+	{
+		[SerializeField]
+		private int damageUnwielded;
+
+		[SerializeField]
+		private int damageWielded;
+
+		[SerializeField]
+		private ItemsSprites Wielded = new ItemsSprites();
+		private ItemsSprites Unwielded = new ItemsSprites();
+
+		[SyncVar(hook = nameof(SyncState))]
+		private bool isWielded;
+
+		private SpriteHandler spriteHandler;
+		private ItemAttributesV2 itemAttributes;
+		private HandsController HandsController;
+
+		public enum hiddenHandValues
+		{
+			bothHands = 0,
+			leftHand = 1,
+			rightHand = 2
+		}
+
+		private void Awake()
+		{
+			itemAttributes = GetComponent<ItemAttributesV2>();
+			spriteHandler = GetComponentInChildren<SpriteHandler>();
+		}
+
+		private void OnEnable()
+		{
+			HandsController.OnSwapHand.AddListener(OnSwapHands);
+		}
+
+		private void OnDisable()
+		{
+			HandsController.OnSwapHand.RemoveListener(OnSwapHands);
+		}
+
+		private void OnSwapHands()
+		{
+			if (isWielded)
+			{
+				Chat.AddExamineMsgFromServer(PlayerManager.LocalPlayerScript.gameObject, $"Your other hand is too busy holding {gameObject.ExpensiveName()}!");
+				HandsController.OnSwapHand.RemoveListener(OnSwapHands);
+				HandsController.SwapHand();
+				HandsController.OnSwapHand.AddListener(OnSwapHands);
+			}
+		}
+
+		private void SyncState(bool oldState, bool newState)
+		{
+			isWielded = newState;
+
+			if (isWielded)
+			{
+				itemAttributes.SetSprites(Wielded);
+			}
+			else
+			{
+				itemAttributes.SetSprites(Unwielded);
+			}
+		}
+
+		public void OnInventoryMoveServer(InventoryMove info)
+		{
+			if (info.InventoryMoveType == InventoryMoveType.Remove)
+			{
+				isWielded = false;
+				ItemSlot hiddenHand = DetermineHiddenHand();
+
+				if (hiddenHand != null)
+				{
+					int hiddenHandSelection = (int) hiddenHandValues.bothHands;
+
+					if (hiddenHand.NamedSlot.GetValueOrDefault(NamedSlot.none) == NamedSlot.leftHand)
+					{
+						hiddenHandSelection = (int) hiddenHandValues.leftHand;
+					}
+					else if (hiddenHand.NamedSlot.GetValueOrDefault(NamedSlot.none) == NamedSlot.rightHand)
+					{
+						hiddenHandSelection = (int) hiddenHandValues.rightHand;
+					}
+
+					HideHand(PlayerManager.LocalPlayerScript.connectionToClient, false, hiddenHandSelection);
+				}
+			}
+		}
+
+		[TargetRpc]
```

Comment:
I'm not too sure, but I think this can be called by any client , so any client could spoof another client's input
@ThatDan123  knows more about it than me

Reply:
This is fine since its Server --> Client.

Only issue is the hands stuff isnt a sync var, so if players leave and rejoin they dont have the correct hand values

Reply:
might be worth making it a  sync var?

Reply:
imma bit confused, is it enough to make SyncVar int to store data about hidden hand and call HideHand every time when player rejoins or do something about HandsController in general so so it remembers which hands are active and which not?

Reply:
Make a sync var in here for the hand enum: https://github.com/unitystation/unitystation/blob/develop/UnityProject/Assets/Scripts/Player/PlayerOnlySyncValues.cs

And on the hook call HideHands (add `none` hidden to the enum, so you can get rid of the bool)

Reply:
only need to change how it's called from server, the hand hiding stuff is handled by dynamic inventory

Reply:
good enough now?